### PR TITLE
Planning: renaming for better semantic understanding

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/common/frame.h [hunk 1]
```diff
@@ -121,11 +121,11 @@ class Frame {
       const double planning_start_time,
       prediction::PredictionObstacles *prediction_obstacles);
 
-  void set_last_planned_trajectory(ADCTrajectory last_planned_trajectory) {
+  void set_current_frame_planned_trajectory(ADCTrajectory last_planned_trajectory) {
```
Comment (xiaoxq) [inline modules/planning/common/frame.h]: Should be either "const ADCTrajectory& last_planned_trajectory" or "ADCTrajectory&& last_planned_trajectory", or else it does several copies.

File: modules/planning/common/frame.h [hunk 2]
```diff
@@ -121,11 +121,11 @@ class Frame {
...
     last_planned_trajectory_ = std::move(last_planned_trajectory);
   }
 
-  ADCTrajectory last_planned_trajectory() const {
+  ADCTrajectory current_frame_planned_trajectory() const {
```
Comment (xiaoxq) [inline modules/planning/common/frame.h]: Return "const ADCTrajectory&". Let the user to decide if they want a copy or a ready only reference.

File: modules/planning/open_space_planning.cc [hunk 3]
```diff
@@ -247,7 +247,7 @@ Status OpenSpacePlanning::Plan(
   Status trajectory_partition_status;
 
   if (status == Status::OK()) {
-    auto trajectory_after_stitching_point = frame_->last_planned_trajectory();
+    auto trajectory_after_stitching_point = frame_->current_frame_planned_trajectory();
```
Comment (xiaoxq) [inline modules/planning/open_space_planning.cc]: For example, this is a copy.

File: modules/planning/tasks/smoothers/smoother.cc [hunk 4]
```diff
@@ -95,7 +95,7 @@ apollo::common::Status Smoother::Smooth(
     AWARN << msg;
     return Status(ErrorCode::PLANNING_ERROR, msg);
   }
-  const auto& previous_planning = previous_frame->last_planned_trajectory();
+  const auto& previous_planning = previous_frame->current_frame_planned_trajectory();
```
Comment (xiaoxq) [inline modules/planning/tasks/smoothers/smoother.cc]: This is also a copy.

File: modules/planning/common/frame.h (hunk 2 above)
Comment (JasonZhou404) [reply modules/planning/common/frame.h]: Yeah, will fix it.

**Pull request merged**